src/c_kzg_4844.c

```diff
@@ -664,6 +656,66 @@ static C_KZG_RET bytes_to_bls_field(fr_t *out, const Bytes32 *b) {
     return C_KZG_OK;
 }
 
+/**
+ * Perform BLS validation required by the types KZGProof and KZGCommitment.
+ *
+ * @param[out]  out The output g1 point
+ * @param[in]   b   The proof/commitment bytes
+ * @retval C_KZG_OK Deserialization successful
+ * @retval C_KZG_BADARGS Invalid input bytes
+ *
+ * @remark This function deviates from the spec because it returns (via an
+ *     output argument) the g1 point. This way is more efficient (faster) but
+ *     the function name is a bit misleading.
+ */
+static C_KZG_RET validate_kzg_g1(g1_t *out, const Bytes48 *b) {
+    /* Fast check without needing to uncompress */
+    if (memcmp(G1_POINT_AT_INFINITY.bytes, b->bytes, sizeof(b)) == 0)
+        return C_KZG_OK;
```

Comment:
Don't we still need to fill `out` in this case?

Reply:
Good catch!

Reply:
Yes, thank you. I'm just going to remove that check, since there's a infinity check below.